src/Features/Core/Diagnostics/BaseDiagnosticIncrementalAnalyzer.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.SolutionCrawler;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    internal abstract class BaseDiagnosticIncrementalAnalyzer : IIncrementalAnalyzer
```

Comment:
I am not sure how it is going to re-use existing incremental analyzer. it won't be able to use current set up of state, serialization or versions. which means it can't use most of existing code.

so only thing it can use is some very basic boiler plate code. 

I think the new prototype should start from clean state. somehow jamming in into current one will create yet another complex thing.